### PR TITLE
Android location options added to the geolocator_android package

### PR DESCRIPTION
**Pull request by florissmit1**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :new: What is the new behavior (if this is a feature change)?
Added the Android location options to the `lib` directory in the geolocator_android package. Moved the `forceAndroidLocationManager` property to the Android Location options instance.

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
Does not apply

### :memo: Links to relevant issues/docs
Does not apply

### :thinking: Checklist before submitting

- [x] I made sure all projects build.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [x] I updated the relevant documentation.
- [x] I rebased onto current `master`.

### CODE REVIEW

File: geolocator_android/lib/android_options.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'package:geolocator_platform_interface/geolocator_platform_interface.dart';
+
+/// Represents different Android specific settings with which you can set a value
+/// other then the default value of the setting.
+class AndroidOptions extends LocationOptions {
```
Comment (mvanbeusekom) [inline geolocator_android/lib/android_options.dart:5]: This should be `AndroidSettings` and extend the `LocationSettings` class from the geolocator_platform_interface package.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Same as with the Apple version, it would be good to create a barrel file which exposes the new types so consumers only have to import one file.

File: geolocator_android/lib/src/types/android_settings.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class AndroidSettings extends LocationSettings {
+  /// Initializes a new [AndroidSpecificSettings] instance with default values.
+  ///
+  /// The following default values are used:
+  /// - forceLocationManager: false
+  AndroidSettings({
+    this.forceLocationManager = false,
+    LocationAccuracy accuracy = LocationAccuracy.best,
+    int distanceFilter = 0,
+    Duration? intervalDuration,
```
Comment (mvanbeusekom) [inline geolocator_android/lib/src/types/android_settings.dart:14]: Shouldn't this also include the `timeLimit` parameter?

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]

File: geolocator_android/lib/src/types/android_settings.dart (hunk 2 above)
Comment (florissmit1) [reply geolocator_android/lib/src/types/android_settings.dart:14]: U're right!

**Pull request merged**